pdns/filterpo.cc

```diff
@@ -27,6 +27,7 @@ bool findNamedPolicy(const map<DNSName, DNSFilterEngine::Policy>& polmap, const
         pol=iter->second;
         return true;
       }
+      s.chopOff(); // So we don't check for *.www.powerdns.com in the first loop.
```

Comment:
Nice catch, but what about just moving the content of the `if(first)` block before the loop?